### ISSUE TITLE
Using pandas 2.1.1 gives warning with altair for apply with convert_dtype

### ISSUE THREAD

**Issue by rikardn** [altair-viz/altair#3197]
pandas 2.1.1 has deprecated the `convert_dtype` option to `apply` triggering a warning from altair:

```
.tox/py311/lib/python3.11/site-packages/altair/utils/core.py:410: in sanitize_dataframe
    col = df[col_name].apply(to_list_if_array, convert_dtype=False)
.tox/py311/lib/python3.11/site-packages/pandas/core/series.py:4753: in apply
    return SeriesApply(
.tox/py311/lib/python3.11/site-packages/pandas/core/apply.py:1170: in __init__
    warnings.warn(
E   FutureWarning: the convert_dtype parameter is deprecated and will be removed in a future version.  Do ``ser.astype(object).apply()`` instead if you want ``convert_dty
pe=False``.
```

**Comment by joelostblom** [comment]
Thanks for the report! This was merged in https://github.com/altair-viz/altair/pull/3191 and will be available in the next release of altair.

**Comment by rikardn** [comment]
Thanks!

**Comment by EwoutH** [comment]
Thanks for fixing this! When is a new release expected that includes this change?

**Comment by mattijn** [comment]
This is already released in version 5.1.2, see https://github.com/altair-viz/altair/releases/tag/v5.1.2
